reference.module

```diff
@@ -322,8 +322,11 @@ function reference_menu() {
  * Provides values for the reference field.
  */
 function reference_query($field_instance, $filter = '', $limit = 0) {
+  // Load field info.
+  $field_info = field_info_field($field_instance['field_name']);
+
   // Determine our target entity parameters based on the field settings.
-  $target_entity_type = $field_instance['entity_type'];
+  $target_entity_type = $field_info['settings']['entity_type'];
```

Comment:
@moonray What is this bug that was fixed? Was it something that wasn't working with the module before you started working on your changes? If so, can you describe it in a separate issue with what was happening so we can reproduce the bug and test that we've fixed it?